src/aspire/utils/rotation.py

```diff
@@ -0,0 +1,300 @@
+"""
+Define a Rotation Class for customized rotation operations used by ASPIRE.
+"""
+
+import numpy as np
+from numpy.linalg import norm
+from scipy.linalg import svd
+from scipy.spatial.transform import Rotation as sp_rot
+
+from aspire.utils import ensure
+from aspire.utils.random import Random
+
+
+class Rotation:
+    def __init__(
+        self, num_rots, seed=None, angles=None, matrices=None, dtype=np.float32
+    ):
+        """
+         Initialize a Rotation object
+
+         :param num_rots: Total number of rotation sets
```

Comment:
Why does this need to be specified separately from `matrices`?

Reply:
For the case of using random angles in `Simulation` object where both `angles` and `matrices` are None.